<?xml version="1.0"?>

<launch>
  <!--Namespace-->
  <arg name="namespace" default="" />
  <!--Server-->
  <arg name="server_port" default="8080" />
  <!--Task parameters-->
  <arg name="max_dist" default="9.0" />
  <arg name="min_dist" default="5.0" />
  <arg name="boey_dist" default="2.5" />
  <arg name="usv_view" default="110" />
  <arg name="boey_cone" default="110" />
  <!--Curriculum-->
  <arg name="curriculum" default="none" />
  <arg name="alpha" default="1" />
  <arg name="warmup" default="0.25e6" />
  <arg name="step_target" default="1.0e6" />

  <!-- STARTS SERVER -->
  <group ns="/$(arg namespace)">
    <group if="$(eval namespace == '')">
      <node pkg="rl_server" name="husky_docking_server" type="ugv_docking_server.py" output="screen">
        <!--Services-->
        <param name="spawn_service" type="string" value="/gazebo/set_model_state" /> 
        <!--Map-->
        <param name="path_to_pose" type="string" value="$(find rl_server)/map_.npy" />
        <!--Robot names-->
        <param name="robot_name" type="string" value="husky" /> 
        <param name="boey_name" type="string" value="boey" /> 
        <!--Server-->
        <param name="server_port" type="int" value="$(arg server_port)" />
        <!--Task parameters-->
        <param name="max_dist" type="double" value="$(arg max_dist)" />
        <param name="min_dist" type="double" value="$(arg min_dist)" />
        <param name="boey_dist" type="double" value="$(arg boey_dist)" />
        <param name="usv_view" type="double" value="$(arg usv_view)" />
        <param name="boey_cone" type="double" value="$(arg boey_cone)" />
        <!--Curriculum-->
        <param name="curriculum" type="string" value="$(arg curriculum)" />
        <param name="alpha" type="double" value="$(arg alpha)" />
        <param name="warmup" type="double" value="$(arg warmup)" />
        <param name="step_target" type="double" value="$(arg step_target)" />
      </node>
    </group>
    <group unless="$(eval namespace == '')">
      <node pkg="rl_server" name="husky_docking_server" type="ugv_docking_server.py" output="screen">
        <!--Services-->
        <param name="spawn_service" type="string" value="/gazebo/set_model_state" /> 
        <!--Map-->
        <param name="path_to_pose" type="string" value="$(find rl_server)/map_.npy" />
        <!--Robot names-->
        <param name="robot_name" type="string" value="$(arg namespace)/husky" /> 
        <param name="boey_name" type="string" value="$(arg namespace)/boey" /> 
        <!--Server-->
        <param name="server_port" type="int" value="$(arg server_port)" />
        <!--Task parameters-->
        <param name="max_dist" type="double" value="$(arg max_dist)" />
        <param name="min_dist" type="double" value="$(arg min_dist)" />
        <param name="boey_dist" type="double" value="$(arg boey_dist)" />
        <param name="usv_view" type="double" value="$(arg usv_view)" />
        <param name="boey_cone" type="double" value="$(arg boey_cone)" />
        <!--Curriculum-->
        <param name="curriculum" type="string" value="$(arg curriculum)" />
        <param name="alpha" type="double" value="$(arg alpha)" />
        <param name="warmup" type="double" value="$(arg warmup)" />
        <param name="step_target" type="double" value="$(arg step_target)" />
      </node>
    </group>
  </group>
</launch>
